internal/models/config.go

```diff
@@ -0,0 +1,171 @@
+corresponding interface and if they have additional settings they must also extend
+the corresponding common settings struct (the easiest approach is to embed the common struct).
+
+For example usage see tests.
+*/
+
+// ConfigV2 defines the configuration V2 for the various elements of collector or agent.
+type ConfigV2 struct {
+	Receivers []ReceiverCfg `mapstructure:"-"`
+	Exporters []ExporterCfg `mapstructure:"-"`
+	Options   []OptionCfg   `mapstructure:"-"`
+	Pipelines Pipelines     `mapstructure:"pipelines"`
+}
+
+// ReceiversKeyName is the configuration key name for receivers section.
+const ReceiversKeyName = "receivers"
+
+// ExportersKeyName is the configuration key name for exporters section.
+const ExportersKeyName = "exporters"
+
+// OptionsKeyName is the configuration key name for options section.
+const OptionsKeyName = "options"
+
+// Pipelines defines the pipelines section of configuration.
+type Pipelines struct {
+	Traces  []*Pipeline `mapstructure:"traces"`
+	Metrics []*Pipeline `mapstructure:"metrics"`
+}
+
+// Pipeline defines a single pipeline
+type Pipeline struct {
+	Name       string              `mapstructure:"name"`
+	Receivers  []PipelineReceiver  `mapstructure:"receivers"`
+	Operations []PipelineOperation `mapstructure:"ordered-operations"`
+	Exporters  []PipelineExporter  `mapstructure:"exporters"`
+}
+
+// PipelineReceiver defines a receiver of a pipeline.
+type PipelineReceiver struct {
+	Name string `mapstructure:"name"`
+}
+
+// PipelineOperation defines an operation of a pipeline.
+type PipelineOperation struct {
+	Name string `mapstructure:"name"`
+}
+
+// PipelineExporter defines an exporter of a pipeline.
+type PipelineExporter struct {
+	Name string `mapstructure:"name"`
+}
+
+// NamedEntity defines a configuration entity that has a name.
+type NamedEntity interface {
+	GetName() string
+}
+
+// ReceiverCfg is the configuration of a receiver. Specific receivers
+// must implement this interface and will typically embed ReceiverCommon
+// struct or a struct that extends it.
+type ReceiverCfg interface {
+	NamedEntity
+}
+
+// TypeKeyName must match Type field key names for Receiver and Exporter.
+const TypeKeyName = "type"
+
+// ExporterCfg is the configuration of an exporter. Specific exporters
+// must implement this interface and will typically embed ExporterCommon
+// struct or a struct that extends it.
+type ExporterCfg interface {
+	NamedEntity
+}
+
+// OptionCfg is the configuration of a processing option. Specific options
+// must implement this interface and will typically embed OptionCommon
+// struct or a struct that extends it.
+type OptionCfg interface {
+	NamedEntity
+}
+
+// Below are common setting structs for Receivers, Exporters and Options.
+// These are helper structs which you can use to implement your specific
+// receiver/exporter/option config storage.
+
+// ReceiverCommon defines common settings for a single-protocol receiver configuration
```

Comment:
nit: missing punctuation at end of this line, same pattern repeats below.

Reply:
Fixed.